cmd/skopeo/utils.go

```diff
@@ -180,12 +184,23 @@ func (opts *imageOptions) newSystemContext() (*types.SystemContext, error) {
 	if opts.credsOption.present && opts.noCreds {
 		return nil, errors.New("creds and no-creds cannot be specified at the same time")
 	}
+	if opts.credsOption.present && opts.userName.present {
+		return nil, errors.New("creds and username cannot be specified at the same time")
+	}
+	if opts.password.present && !opts.userName.present {
```

Comment:
Should `--username` without `--password` be accepted? I don’t have a _strong_ opinion, but…

Reply:
That’s a good idea — but also a new feature we don’t yet have, and I don’t think this PR (targeted at allowing `:` in usernames) should block on implementing the prompting.

So for now, we should probably reject that combination, so that we can add prompting later.